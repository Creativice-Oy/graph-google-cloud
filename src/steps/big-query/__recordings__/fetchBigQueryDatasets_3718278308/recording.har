{
  "log": {
    "_recordingName": "fetchBigQueryDatasets",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "4.3.0"
    },
    "entries": [
      {
        "_id": "acea721c8193b51ced888cae721cc423",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 745,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "content-type",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "content-length",
              "value": "745"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "www.googleapis.com"
            }
          ],
          "headersSize": 283,
          "httpVersion": "HTTP/1.1",
          "method": "POST",
          "postData": {
            "mimeType": "application/json",
            "params": [],
            "text": "[REDACTED]"
          },
          "queryString": [],
          "url": "https://www.googleapis.com/oauth2/v4/token"
        },
        "response": {
          "bodySize": 571,
          "content": {
            "mimeType": "application/json; charset=UTF-8",
            "size": 571,
            "text": "{\"access_token\":\"[REDACTED]\",\"expires_in\":9999,\"token_type\":\"Bearer\"}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Mon, 18 Jan 2021 14:10:46 GMT"
            },
            {
              "name": "server",
              "value": "scaffolding on HTTPServer2"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3-29=\":443\"; ma=2592000,h3-T051=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\""
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 511,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-01-18T14:10:46.772Z",
        "time": 202,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 202
        }
      },
      {
        "_id": "e0cca21e28ce3079b9c7b13ec5304390",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "x-goog-api-client",
              "value": "gdcl/4.4.3 gl-node/14.15.4 auth/6.0.5"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "google-api-nodejs-client/4.4.3 (gzip)"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "cloudresourcemanager.googleapis.com"
            }
          ],
          "headersSize": 566,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://cloudresourcemanager.googleapis.com/v1/projects/j1-gc-integration-dev-300716"
        },
        "response": {
          "bodySize": 462,
          "content": {
            "mimeType": "application/json; charset=UTF-8",
            "size": 462,
            "text": "{\"projectNumber\":\"165882964161\",\"projectId\":\"j1-gc-integration-dev-300716\",\"lifecycleState\":\"ACTIVE\",\"name\":\"j1-gc-integration-dev\",\"createTime\":\"2021-01-04T16:56:24.926Z\",\"parent\":{\"type\":\"organization\",\"id\":\"590372371675\"}}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Mon, 18 Jan 2021 14:10:47 GMT"
            },
            {
              "name": "server",
              "value": "ESF"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "server-timing",
              "value": "gfet4t7; dur=179"
            },
            {
              "name": "alt-svc",
              "value": "h3-29=\":443\"; ma=2592000,h3-T051=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\""
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 521,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-01-18T14:10:46.986Z",
        "time": 364,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 364
        }
      },
      {
        "_id": "acea721c8193b51ced888cae721cc423",
        "_order": 1,
        "cache": {},
        "request": {
          "bodySize": 745,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "content-type",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "content-length",
              "value": "745"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "www.googleapis.com"
            }
          ],
          "headersSize": 283,
          "httpVersion": "HTTP/1.1",
          "method": "POST",
          "postData": {
            "mimeType": "application/json",
            "params": [],
            "text": "[REDACTED]"
          },
          "queryString": [],
          "url": "https://www.googleapis.com/oauth2/v4/token"
        },
        "response": {
          "bodySize": 595,
          "content": {
            "mimeType": "application/json; charset=UTF-8",
            "size": 595,
            "text": "{\"access_token\":\"[REDACTED]\",\"expires_in\":9999,\"token_type\":\"Bearer\"}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Mon, 18 Jan 2021 14:10:47 GMT"
            },
            {
              "name": "server",
              "value": "scaffolding on HTTPServer2"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3-29=\":443\"; ma=2592000,h3-T051=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\""
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 511,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-01-18T14:10:47.367Z",
        "time": 194,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 194
        }
      },
      {
        "_id": "14aa4c3953be992452729cdb5994a8a8",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "x-goog-api-client",
              "value": "gdcl/4.4.3 gl-node/14.15.4 auth/6.0.5"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "google-api-nodejs-client/4.4.3 (gzip)"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "bigquery.googleapis.com"
            }
          ],
          "headersSize": 560,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://bigquery.googleapis.com/bigquery/v2/projects/j1-gc-integration-dev-300716/datasets"
        },
        "response": {
          "bodySize": 461,
          "content": {
            "mimeType": "application/json; charset=UTF-8",
            "size": 461,
            "text": "{\"kind\":\"bigquery#datasetList\",\"etag\":\"qHMBRilsO2iq1YyXCsjCxQ==\",\"datasets\":[{\"kind\":\"bigquery#dataset\",\"id\":\"j1-gc-integration-dev-300716:test_big_query_dataset\",\"datasetReference\":{\"datasetId\":\"test_big_query_dataset\",\"projectId\":\"j1-gc-integration-dev-300716\"},\"location\":\"US\"}]}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "etag",
              "value": "qHMBRilsO2iq1YyXCsjCxQ=="
            },
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Mon, 18 Jan 2021 14:10:48 GMT"
            },
            {
              "name": "server",
              "value": "ESF"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3-29=\":443\"; ma=2592000,h3-T051=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\""
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 520,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-01-18T14:10:47.566Z",
        "time": 556,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 556
        }
      },
      {
        "_id": "5017484b86396a35700ec9d3b8215410",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "x-goog-api-client",
              "value": "gdcl/4.4.3 gl-node/14.15.4 auth/6.0.5"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "google-api-nodejs-client/4.4.3 (gzip)"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "bigquery.googleapis.com"
            }
          ],
          "headersSize": 583,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://bigquery.googleapis.com/bigquery/v2/projects/j1-gc-integration-dev-300716/datasets/test_big_query_dataset"
        },
        "response": {
          "bodySize": 836,
          "content": {
            "mimeType": "application/json; charset=UTF-8",
            "size": 836,
            "text": "{\"kind\":\"bigquery#dataset\",\"etag\":\"Q1RJczpo1aoPRRoUJ1pElQ==\",\"id\":\"j1-gc-integration-dev-300716:test_big_query_dataset\",\"selfLink\":\"https://www.googleapis.com/bigquery/v2/projects/j1-gc-integration-dev-300716/datasets/test_big_query_dataset\",\"datasetReference\":{\"datasetId\":\"test_big_query_dataset\",\"projectId\":\"j1-gc-integration-dev-300716\"},\"access\":[{\"role\":\"WRITER\",\"specialGroup\":\"allAuthenticatedUsers\"},{\"role\":\"WRITER\",\"specialGroup\":\"projectWriters\"},{\"role\":\"OWNER\",\"specialGroup\":\"projectOwners\"},{\"role\":\"OWNER\",\"userByEmail\":\"admin@creativice.com\"},{\"role\":\"READER\",\"iamMember\":\"allUsers\"},{\"role\":\"READER\",\"specialGroup\":\"projectReaders\"}],\"creationTime\":\"1610919167542\",\"lastModifiedTime\":\"1610919344642\",\"location\":\"US\"}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "etag",
              "value": "Q1RJczpo1aoPRRoUJ1pElQ=="
            },
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Mon, 18 Jan 2021 14:10:48 GMT"
            },
            {
              "name": "server",
              "value": "ESF"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3-29=\":443\"; ma=2592000,h3-T051=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\""
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 520,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-01-18T14:10:48.127Z",
        "time": 574,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 574
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
