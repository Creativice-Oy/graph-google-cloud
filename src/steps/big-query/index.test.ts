import { createMockStepExecutionContext } from '@jupiterone/integration-sdk-testing';
import { fetchBigQueryDatasets, fetchBigQueryTables } from '.';
import { integrationConfig } from '../../../test/config';
import { Recording, setupGoogleCloudRecording } from '../../../test/recording';
import { IntegrationConfig } from '../../types';
import { fetchKmsCryptoKeys, fetchKmsKeyRings } from '../kms';
import {
  BIG_QUERY_DATASET_ENTITY_TYPE,
  RELATIONSHIP_TYPE_DATASET_USES_KMS_CRYPTO_KEY,
  BIG_QUERY_TABLE_ENTITY_TYPE,
  RELATIONSHIP_TYPE_DATASET_HAS_TABLE,
} from './constants';

describe('#fetchBigQueryDatasets', () => {
  let recording: Recording;

  beforeEach(() => {
    recording = setupGoogleCloudRecording({
      directory: __dirname,
      name: 'fetchBigQueryDatasets',
    });
  });

  afterEach(async () => {
    await recording.stop();
  });

  test('should collect data', async () => {
    const context = createMockStepExecutionContext<IntegrationConfig>({
      instanceConfig: integrationConfig,
    });

    await fetchKmsKeyRings(context);
    await fetchKmsCryptoKeys(context);
    await fetchBigQueryDatasets(context);

    expect({
      numCollectedEntities: context.jobState.collectedEntities.length,
      numCollectedRelationships: context.jobState.collectedRelationships.length,
      collectedEntities: context.jobState.collectedEntities,
      collectedRelationships: context.jobState.collectedRelationships,
      encounteredTypes: context.jobState.encounteredTypes,
    }).toMatchSnapshot();

    expect(
      context.jobState.collectedEntities.filter(
        (e) => e._type === BIG_QUERY_DATASET_ENTITY_TYPE,
      ),
    ).toMatchGraphObjectSchema({
      _class: ['DataStore', 'Database'],
      schema: {
        additionalProperties: false,
        properties: {
          _type: { const: 'google_bigquery_dataset' },
          _rawData: {
            type: 'array',
            items: { type: 'object' },
          },
          name: { type: 'string' },
          description: { type: 'string' },
          public: { type: 'boolean' },
          kmsKeyName: { type: 'string' },
          location: { type: 'string' },
          encrypted: { type: 'boolean' },
          classification: { const: null },
          etag: { type: 'string' },
          createdOn: { type: 'number' },
          updatedOn: { type: 'number' },
          webLink: { type: 'string' },
        },
      },
    });

    expect(
      context.jobState.collectedRelationships.filter(
        (e) => e._type === RELATIONSHIP_TYPE_DATASET_USES_KMS_CRYPTO_KEY,
      ),
    ).toMatchDirectRelationshipSchema({
      schema: {
        properties: {
          _class: { const: 'USES' },
          _type: {
            const: 'google_bigquery_dataset_uses_kms_crypto_key',
          },
        },
      },
    });
  });
});

describe('#fetchBigQueryTables', () => {
  let recording: Recording;

  beforeEach(() => {
    recording = setupGoogleCloudRecording({
      directory: __dirname,
      name: 'fetchBigQueryTables',
    });
  });

  afterEach(async () => {
    await recording.stop();
  });

  test('should collect data', async () => {
    const context = createMockStepExecutionContext<IntegrationConfig>({
      instanceConfig: integrationConfig,
    });

    await fetchBigQueryDatasets(context);
    await fetchBigQueryTables(context);

    expect({
      numCollectedEntities: context.jobState.collectedEntities.length,
      numCollectedRelationships: context.jobState.collectedRelationships.length,
      collectedEntities: context.jobState.collectedEntities,
      collectedRelationships: context.jobState.collectedRelationships,
      encounteredTypes: context.jobState.encounteredTypes,
    }).toMatchSnapshot();

    expect(
      context.jobState.collectedEntities.filter(
        (e) => e._type === BIG_QUERY_TABLE_ENTITY_TYPE,
      ),
    ).toMatchGraphObjectSchema({
      _class: ['DataObject'],
      schema: {
        additionalProperties: false,
        properties: {
          _type: { const: 'google_bigquery_table' },
          _rawData: {
            type: 'array',
            items: { type: 'object' },
          },
          name: { type: 'string' },
          public: { type: 'boolean' },
          type: { type: 'string' },
          friendlyName: { type: 'string' },
          classification: { const: null },
          webLink: { type: 'string' },
          createdOn: { type: 'number' },
          expirationTime: { type: 'number' },
        },
      },
    });

    expect(
      context.jobState.collectedRelationships.filter(
        (e) => e._type === RELATIONSHIP_TYPE_DATASET_HAS_TABLE,
      ),
    ).toMatchDirectRelationshipSchema({
      schema: {
        properties: {
          _class: { const: 'HAS' },
          _type: {
            const: 'google_bigquery_dataset_has_table',
          },
        },
      },
    });
  });
});
