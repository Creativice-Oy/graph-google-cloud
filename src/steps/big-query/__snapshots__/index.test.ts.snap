// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#fetchBigQueryDatasets should collect data 1`] = `
Object {
  "collectedEntities": Array [
    Object {
      "_class": Array [
        "Account",
      ],
      "_key": "j1-gc-integration-dev-300716",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "createTime": "2021-01-04T16:56:24.926Z",
            "lifecycleState": "ACTIVE",
            "name": "j1-gc-integration-dev",
            "parent": Object {
              "id": "590372371675",
              "type": "organization",
            },
            "projectId": "j1-gc-integration-dev-300716",
            "projectNumber": "165882964161",
          },
        },
      ],
      "_type": "google_cloud_project",
      "createdOn": 1609779384926,
      "displayName": "j1-gc-integration-dev",
      "lifecycleState": "ACTIVE",
      "name": "j1-gc-integration-dev",
      "parent.id": "590372371675",
      "parent.type": "organization",
      "projectNumber": "165882964161",
    },
    Object {
      "_class": Array [
        "DataStore",
      ],
      "_key": "j1-gc-integration-dev-300716:test_big_query_dataset",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "access": Array [
              Object {
                "role": "WRITER",
                "specialGroup": "allAuthenticatedUsers",
              },
              Object {
                "role": "WRITER",
                "specialGroup": "projectWriters",
              },
              Object {
                "role": "OWNER",
                "specialGroup": "projectOwners",
              },
              Object {
                "role": "OWNER",
                "userByEmail": "admin@creativice.com",
              },
              Object {
                "iamMember": "allUsers",
                "role": "READER",
              },
              Object {
                "role": "READER",
                "specialGroup": "projectReaders",
              },
            ],
            "creationTime": "1610919167542",
            "datasetReference": Object {
              "datasetId": "test_big_query_dataset",
              "projectId": "j1-gc-integration-dev-300716",
            },
            "etag": "Q1RJczpo1aoPRRoUJ1pElQ==",
            "id": "j1-gc-integration-dev-300716:test_big_query_dataset",
            "kind": "bigquery#dataset",
            "lastModifiedTime": "1610919344642",
            "location": "US",
            "selfLink": "https://www.googleapis.com/bigquery/v2/projects/j1-gc-integration-dev-300716/datasets/test_big_query_dataset",
          },
        },
      ],
      "_type": "google_bigquery_dataset",
      "classification": null,
      "createdOn": undefined,
      "displayName": "test_big_query_dataset",
      "encrypted": true,
      "etag": "Q1RJczpo1aoPRRoUJ1pElQ==",
      "id": "j1-gc-integration-dev-300716:test_big_query_dataset",
      "location": "US",
      "name": "test_big_query_dataset",
      "public": true,
      "webLink": "https://www.googleapis.com/bigquery/v2/projects/j1-gc-integration-dev-300716/datasets/test_big_query_dataset",
    },
  ],
  "collectedRelationships": Array [],
  "encounteredTypes": Array [
    "google_cloud_project",
    "google_bigquery_dataset",
  ],
  "numCollectedEntities": 2,
  "numCollectedRelationships": 0,
}
`;
